Add warning about failed AccountDelete costs

mDuo13 opened the pull request:
Fixes #850.

Note, due to the nature of consensus, `fail_hard` is not a failsafe guarantee that the transaction won't be validated with a `tec` code, but it prevents the vast majority of cases.

mDuo13 committed:
Add warning about failed AccountDelete costs

Fixes #850.

ximinez commented on content/concepts/payment-system-basics/accounts/accounts.md:
```diff
@@ -75,6 +75,8 @@ To be deleted, an account must meet the following requirements:
 
 After an account has been deleted, it can be re-created in the ledger through the normal method of [creating accounts](#creating-accounts). An account that has been deleted and re-created is no different than an account that has been created for the first time.
 
+**Warning:** The [AccountDelete transaction][]'s transaction cost applies even if the transaction fails because the account does not meet the requirements to be deleted. To avoid paying 5 XRP in case account deletion fails, [submit the transaction](submit.html) with `fail_hard` enabled so that the transaction is rejected unless it is provisionally successful.
```
Suggestion:

**Warning**: The [AccountDelete transaction][]'s cost applies if the transaction is properly structured and signed, but the account does not meet the requirements to be deleted. [Submit the transaction](submit.html) with `fail_hard` enabled to lower the chances of paying the high transaction fee if the account is not deletable when the transaction is validated.

mDuo13 committed:
Deletion tx cost: edit descripion & add to AcctDel tx ref too

mDuo13 commented:
Rebased on master to get some broken link fixes, reworded the sentence @ximinez pointed out, and added a similar warning to the AccountDelete tx reference page.

ximinez commented on content/concepts/payment-system-basics/accounts/accounts.md:
```diff
@@ -75,6 +75,8 @@ To be deleted, an account must meet the following requirements:
 
 After an account has been deleted, it can be re-created in the ledger through the normal method of [creating accounts](#creating-accounts). An account that has been deleted and re-created is no different than an account that has been created for the first time.
 
+**Warning:** The [AccountDelete transaction][]'s transaction cost always applies when the transaction is included in a validated ledger, even if the transaction failed because the account does not meet the requirements to be deleted. To greatly reduce the chances of paying the high transaction cost if the account cannot be deleted, [submit the transaction](submit.html) with `fail_hard` enabled.
```
Would it be worth mentioning the `"deletion_blockers_only" : "true"` to the `account_objects` method here?

ximinez reviewed (approved):
I had another thought, but I'm not going to hold up this change if you don't think it's worth it.

MarkusTeufelberger commented:
Maybe also mention how to check each requirement with the respective API calls?

mDuo13 commented:
I would like to add a more detailed tutorial with step-by-step instructions on deleting your account, including how to check the requirements, etc., but I think that's a bigger scope and should not hold up this PR.

mDuo13 commented:
The broken links are some kind of GitHub or link checker glitch. The links from this PR work.
